hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -351,6 +373,19 @@ void deleteJob(long jobId) {
         // delete the job record and related records
         jobExecutionRecords.remove(jobId);
         jobRecords.remove(jobId);
+        deleteTempFiles("jet-" + instance.getName() + "-" + idToString(jobId));
+    }
+
+    private void deleteTempFiles(String filter) {
```

Comment:
This seems rather expensive - it's going to iterate all temp files each time a job completes? We made a lot of optimizations for jobs that take just a few ms (small query jobs). It could be done as part of the Job GC process. Furthermore this is only called on master node, not on other nodes, how are those files on other nodes cleaned up?